rotkehlchen/tasks/manager.py

```diff
@@ -397,25 +406,47 @@ def _maybe_decode_evm_transactions(self) -> None:
     def _maybe_check_premium_status(self) -> None:
         """
         Validates the premium status of the account and if the credentials are not valid
-        it deactivates the user's premium status.
+        it retries 3 times before deactivating the user's premium status. If the
+        credentials are valid and the premium status is not correct it will reactivate
+        the user's premium status.
         """
         now = ts_now()
         if now - self.last_premium_status_check < PREMIUM_STATUS_CHECK:
             return
 
         db_credentials = self.database.get_rotkehlchen_premium()
-        if db_credentials:
-            try:
-                premium_create_and_verify(db_credentials)
-            except PremiumAuthenticationError as e:
-                message = (
-                    f'Could not authenticate with the rotkehlchen server with '
-                    f'the API keys found in the Database. Error: {str(e)}. Will '
-                    f'deactivate the premium status.'
+        if not db_credentials:
```

Comment:
nitpick: We use explicit checks for truthfallness and existence of None.

```suggestion
        if db_credentials is None:
```

Reply:
This is still not resolved. Why have you not done this and why is it marked as resolved?